addons/account/models/account_invoice.py

```diff
@@ -253,11 +253,11 @@ def _compute_payments(self):
     move_name = fields.Char(string='Journal Entry Name', readonly=False,
         default=False, copy=False,
         help="Technical field holding the number given to the invoice, automatically set when the invoice is validated then stored to set the same number again if the invoice is cancelled, set to draft and re-validated.")
-    reference = fields.Char(string='Vendor Reference', copy=False,
+    reference = fields.Char(string='Payment Ref.', compute='_compute_reference', copy=False, store=True,
```

Comment:
cannot be a computed field. It needs to be a normal field, that can be manually set, upon validation of a customer invoice (only) it will be filled if not already existing